scripts/wrapper/withdraw.js

```diff
@@ -0,0 +1,116 @@
+module.exports = function(web3, artifacts) {
+  const fs = require("fs").promises
+
+  const { fromErc20Units, shortenedAddress } = require("../utils/printing_tools")
+  const {
+    getExchange,
+    fetchTokenInfoForFlux,
+    buildRequestWithdraw,
+    buildWithdraw,
+    buildTransferFundsToMaster,
+    buildWithdrawAndTransferFundsToMaster,
+  } = require("../utils/trading_strategy_helpers")(web3, artifacts)
+
+  const assertGoodArguments = function(argv) {
+    if (!argv.requestWithdraw && !argv.withdraw && !argv.transferFundsToMaster) {
+      throw new Error("Argument error: one of --requestWithdraw, --withdraw, --transferFundsToMaster must be given")
+    } else if (argv.requestWithdraw && (argv.transferFundsToMaster || argv.withdraw)) {
+      throw new Error("Argument error: --requestWithdraw cannot be used with any of --withdraw, --transferFundsToMaster")
+    }
+  }
+
+  const getMaxWithdrawableAmount = async function(argv, bracketAddress, tokenInfo, exchange, printOutput = false) {
+    const log = printOutput ? (...a) => console.log(...a) : () => {}
+    let amount
+    const token = tokenInfo.instance
+    if (argv.requestWithdraw) amount = (await exchange.getBalance(bracketAddress, tokenInfo.address)).toString()
+    else if (argv.withdraw) {
+      const currentBatchId = Math.floor(Date.now() / (5 * 60 * 1000)) // definition of BatchID, it avoids making a web3 request for each withdrawal to get BatchID
+      const pendingWithdrawal = await exchange.getPendingWithdraw(bracketAddress, tokenInfo.address)
+      if (pendingWithdrawal[1].toNumber() == 0) {
+        log("Warning: no withdrawal was requested for address", bracketAddress, "and token", tokenInfo.symbol)
+        amount = "0"
+      }
+      if (amount != "0" && pendingWithdrawal[1].toNumber() >= currentBatchId) {
+        log("Warning: amount cannot be withdrawn from the exchange right now, withdrawing zero")
+        amount = "0"
+      }
+      amount = pendingWithdrawal[0].toString()
```

Comment:
It sucks that we don't have a proper way to determine the withdrawal balance in the smart contract. But it should be the max amount of 
```
 max((await token.balanceOf(bracketAddress)).toString(), pendingWithdrawal[0].toString())
```

(just noting this here, I guess its anyways just copy paste code)

Reply:
You are saying that the user might get less than the amount in the pending withdrawal, and thus the multisend withdraw+transferFundsToMaster would fail, right? This is a very good point, the fix is going to be ugly however. The only way I see is to use `web3.getStorageAt` and manually compute the storage position of `balanceState[braketAddress]`. Alternatively, we give up on the bundled transaction and we make the user do withdraw and transferFundsToMaster separately, so a grand total of three transactions to withdraw.

Reply:
Exactly,

lets go for the `web3.getStorageAt` route, as longterm we want to allow more and more users to use these liquidity provision scripts. (maybe even with front-end support)

let me create an issue for that.